build/configs/qemu/scripts/flash-sram.ld

```diff
@@ -114,6 +114,9 @@ SECTIONS
 		*(.gnu.linkonce.b.*)
 		*(COMMON)
 		_ebss = ABSOLUTE(.);
+		_sidle_stack = ABSOLUTE(.);
+		/* Heap start address in internal RAM */
+		_sint_heap_start = ABSOLUTE(.);
```

Comment:
It looks wrong. `_sidle_stack` and `_sint_heap_start` can't be the same. Because `_sint_heap_start` should be `_sidle_stack + CONFIG_IDLETHREAD_STACKSIZE`.

Reply:
This is added in the start.c of chipsets as below:
#define HEAP_BASE ((uintptr_t)&_sint_heap_start + CONFIG_IDLETHREAD_STACKSIZE)

Reply:
When someone sets the same address for both _sidle_stack and _sint_heap_start, then this calculation is correct. 
But
1. logically it's wrong, idle stack and heap start can't be the same.
2. If idle stack and internal heap is separate, then this calculation is wrong. At that case, the HEAP_BASE should be just _sint_heap_start.